File: VideoGameBacklog/Models/ProgressLog.cs
using System;
using System.Collections.Generic;

namespace VideoGameBacklog.Models;

public partial class ProgressLog
{
    public int LogId { get; set; }

    public int? UserId { get; set; }

    public int? GameId { get; set; }

    public string? Status { get; set; }

    public int? PlayTime { get; set; }

    public virtual Game? Game { get; set; }

    public virtual User? User { get; set; }
}

File: BacklogFrontend/src/app/models/progresslog.ts
import { GameAPI } from "./game";
import { User } from "./user";

export interface ProgressLog {
    logID: number;
    userID: number;
    gameID: number;
    status: string;
    playtime: string;
    user: User;
}

export interface BackLogDTO {
    userId: number;
    gameId: number;
    status: string;
    playTime: number;
}

export interface RetrieveBackLogDTO {
    status: string;
    playTime: number;
    game: GameAPI;
}